### ISSUE TITLE
CherryTree 037.2-1 Crashing in Manjaro Linux

### ISSUE THREAD

**Issue by sueridgepipe** [giuspen/cherrytree#179]
On application launch CherryTree crashes immediately.  The icon appears in the task manager, a cherry with a rotating circle animation, then disappears after 1-2 seconds.

I am running the latest Manjaro Linux Stable release (2016-09-13), CherryTree was working perfectly prior to the upgrade.  All other apps work perfectly, but CherryTree doesn't even load.  

Does CherryTree cut a log file that would assist in debugging this crash?

Is there a way to access the info in the notes ctx file without the app?   I have a lot of info in there that I can no longer access.

Any help would be appreciated.... thanks.

**Comment by sueridgepipe** [comment]
Attempting to run CherryTree from CLI I get...

```
$ cherrytree

(process:12237): Gtk-WARNING **: Locale not supported by C library.
        Using the fallback 'C' locale.
Traceback (most recent call last):
  File "/usr/bin/cherrytree", line 72, in <module>
    f_main(args)
  File "/usr/bin/cherrytree", line 50, in f_main
    main.main(args)
  File "/usr/share/cherrytree/modules/main.py", line 218, in main
    lang_str = initializations()
  File "/usr/share/cherrytree/modules/main.py", line 151, in initializations
    locale.setlocale(locale.LC_ALL, '')
  File "/usr/lib/python2.7/locale.py", line 581, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```

Locale hasn't changed since OS installation, not sure what is going on here...

**Comment by giuspen** [comment]
Enter the settings dir
/home/YOURUSERNAME/.config/cherrytree/
Try to delete the lang file and run again from the command line

**Comment by sueridgepipe** [comment]
```
$ cat config.cfg
[state]
file_dir = /home/manjaro/Documents
file_name = Notes.ctx
toolbar_visible = True
win_is_maximized = False
win_position_x = 374
win_position_y = 41
win_size_w = 1087
win_size_h = 960
hpaned_pos = 193
node_path = 2 0
cursor_position = 452
recent_docs = L2hvbWUvZ2xlbm4vRG9jdW1lbnRzL05vdGVzLmN0eA==
pick_dir = 
link_type = webs
show_node_name_header = True
nodes_on_node_name_header = 3
toolbar_icon_size = 2
bg = #ffffffffffff
                                                                                                           
[tree]                                                                                                     
rest_exp_coll = 1                                                                                          
nodes_bookm_exp = True                                                                                     
nodes_icons = c                                                                                            
default_icon_text = 25                                                                                     
tree_right_side = False                                                                                    
cherry_wrap_width = 130                                                                                    
tree_click_focus_text = False                                                                              
tree_click_expand = True                                                                                   
                                                                                                           
[editor]                                                                                                   
syntax_highlighting = custom-colors                                                                        
auto_syn_highl = sh                                                                                        
style_scheme = oblivion                                                                                    
spell_check_lang = en_US                                                                                   
enable_spell_check = False                                                                                 
show_line_numbers = False                                                                                  
spaces_instead_tabs = True                                                                                 
tabs_width = 4                                                                                             
anchor_size = 16
embfile_size = 48
line_wrapping = True
auto_smart_quotes = True
wrapping_indent = -14
auto_indent = True
rt_show_white_spaces = False
pt_show_white_spaces = False
rt_highl_curr_line = False
pt_highl_curr_line = False
space_around_lines = 0
relative_wrapped_space = 50
h_rule = ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
special_chars = “”„‘’•◇▪▸☐☑☒★…‰€©®™°↓↑→←↔↵⇓⇑⇒⇐⇔»«▼▲►◄≤≥≠≈±¹²³½¼⅛×÷∞ø∑√∫ΔδΠπΣΦΩωαβγεηλμ☺☻☼♥♣♦✔♀♂♪♫✝
selword_chars = .-@
chars_listbul = •◇▪-→⇒
chars_toc = ▸•◇▪
latest_tag_prop = weight
latest_tag_val = heavy
timestamp_format = %Y/%m/%d - %H:%M
links_underline = True
links_relative = False
weblink_custom_action = Falsefirefox %s &
filelink_custom_action = Falsexdg-open %s &
folderlink_custom_action = Falsexdg-open %s &

[codebox]
codebox_width = 700.0
codebox_height = 100.0
codebox_width_pixels = True
codebox_line_num = False
codebox_match_bra = True
codebox_syn_highl = plain-text
codebox_auto_resize = True

[table]
table_rows = 3
table_columns = 3
table_column_mode = rename
table_col_min = 40
table_col_max = 60

[fonts]
text_font = Noto Sans 11
tree_font = Noto Sans 10
code_font = Noto Mono 11

[colors]
rt_def_fg = #000000
rt_def_bg = #ffffff
tt_def_fg = #0b0c0c
tt_def_bg = #e0e0e0
palette_list = #ffffff:#000000:#7f7f7f:#ff0000:#a020f0:#0000ff:#add8e6:#00ff00:#ffff00:#ffa500:#e6e6fa:#a52a2a:#8b6914:#1e90ff:#ffc0cb:#90ee90:#1a1a1a:#4d4d4d:#bfbfbf:#e5e5e5
col_link_webs = #00004444ffff
col_link_node = #071c838e071c
col_link_file = #8b8b69691414
col_link_fold = #7f7f7f7f7f7f

[misc]
toolbar_ui_list = tree_add_node,tree_add_subnode,separator,go_node_prev,go_node_next,separator,*,ct_save,export_pdf,separator,find_in_allnodes,separator,handle_bull_list,handle_num_list,handle_todo_list,separator,handle_image,handle_table,handle_codebox,handle_embfile,handle_link,handle_anchor,separator,fmt_rm,fmt_color_fg,fmt_color_bg,fmt_bold,fmt_italic,fmt_underline,fmt_strikethrough,fmt_h1,fmt_h2,fmt_h3,fmt_small,fmt_superscript,fmt_subscript,fmt_monospace
systray = False
start_on_systray = False
use_appind = False
autosave = False
autosave_val = 5
check_version = False
reload_doc_last = True
mod_time_sent = True
backup_copy = False
backup_num = 1
autosave_on_quit = True
limit_undoable_steps = 20

[keyboard]

```

**Comment by sueridgepipe** [comment]
By LANG file do you mean locale.conf file?

**Comment by giuspen** [comment]
You can wipe out the whole content of the folder and then run cherrytree
and see if it starts. If starts empty file-open and spot your document
Notes.ctx in the Documents folder (according to your settings). You should
upgrade to latest 0.37.4 though

**Comment by sueridgepipe** [comment]
0.37.2 is what is currently in the Manjaro repository...  how would the repo maintainers and distro devs be notified of updated versions?

Trashed the ~/.config/cherrytree directory and still the same cli error.

**Comment by giuspen** [comment]
Ok try to download the 0.37.4.tar.xz from the cherrytree home, extract,
open a terminal in the extracted folder and run
./cherrytree

**Comment by sueridgepipe** [comment]
Same error...

```
$ ./cherrytree

(process:11684): Gtk-WARNING **: Locale not supported by C library.
        Using the fallback 'C' locale.
Traceback (most recent call last):
  File "./cherrytree", line 72, in <module>
    f_main(args)
  File "./cherrytree", line 50, in f_main
    main.main(args)
  File "/home/glenn/Downloads/cherrytree-0.37.4/modules/main.py", line 218, in main
    lang_str = initializations()
  File "/home/glenn/Downloads/cherrytree-0.37.4/modules/main.py", line 151, in initializations
    locale.setlocale(locale.LC_ALL, '')
  File "/usr/lib/python2.7/locale.py", line 581, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```
This is very unusual, bizarre it is only happening with CherryTree and no other apps.

Cannot find the default locale...

```
$ locale -a
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_COLLATE to default locale: No such file or directory
C
POSIX
en_AU
en_AU.iso88591
en_AU.utf8
en_US
en_US.iso88591
en_US.utf8
```

**Comment by giuspen** [comment]
Please be sure that you were running "./cherrytree" and not "cherrytree"
because the second would still run the installed and not the local.
I can try to install Manjaro on a virtual machine and see if I have the
same issue but not sure I can do it today. What is the Manjaro version?

**Comment by sueridgepipe** [comment]
Yeah, made sure of that, and included it in the output as evidence.

Latest KDE Manjaro ISO is 16.08 I think, and there was an update on 2016.09.13.  Another Manjaro user installed cherrytree as a test about an hour ago and no issues.

Does glibc provide the locale parameter info for the runtime errors listed above?  Is this info sourced from files in the OS?  Maybe these files are not present for en_AU locale?  This is driving me nuts.

Any way of accessing the data in Notes.ctx without the app?

**Comment by giuspen** [comment]
You can access the data without cherrytree you have to use 7zip to extract
the password protected archive .ctx and then use sqlite to look at the data
of the extracted .ctb

I advise you to look at the dependencies of the manjaro cherrytree package
and reinstall the dependencies starting from gtk2 and python-gtk2

**Comment by sueridgepipe** [comment]
Thanks for your help.

For some reason it looks like python cannot determine the locale at runtime.

Within my local python runtime environment
```
import locale
locale.getlocale()
```
Returns
```
(None,None)
```
python2-dbus seems like a good place to start I think.  This update did have other issues associated with d-bus.

**Comment by sueridgepipe** [comment]
This issue is now solved.

The issue was that locale settings were not being synched correctly between Manjaro and KDE.  So whilst they were correct withing Manjaro (ie locale.conf, locale.gen, etc), they were wrong in KDE.

I changed ~/.config/plasma-localerc and ~/.config/plasma-locale-settings to the correct locales and is now working.

Python proof of concept after fix...
```
>>> import locale
>>> locale.getlocale()
('en_AU', 'UTF-8')
```
CherryTree now works, I'll let the repo maintainers know they are a couple of version behind, although this could be an Arch mothership issue.

Thanks for your help, much appreciated.  Thanks also for a great app, I use it everyday, it is invaluable for people like me with a bad memory (head trauma).
